fix(feature/improving-code-quality): Improved code quality

saadqu opened the pull request:
- added base service to extend on for data service and other services
- rewritten form validation messages into a seperate component to reuse
- renamed service calls to a more logic name

saadqu committed:
fix(feature/improving-code-quality): Improved code quality

- added base service to extend on for data service and other services
- rewritten form validation messages into a seperate component to reuse
- renamed service calls to a more logic name

ismail-codinglab commented on src/app/form-validation/form-validation.component.html:
```diff
@@ -0,0 +1,14 @@
+<div *ngIf="control.invalid && control.touched" class="alert alert-warning pt-2">
```
Remove 'pt-2' it does nothing

- [x] removed 'pt-2'

saadqu replied:
Ok

saadqu replied:
Done

ismail-codinglab reviewed (commented):
Please check

saadqu committed:
fix(feature/mproving-code-quality):
removed pt-2

removed the pt-2 class from the form-validation div, because it was not doing anything

ismail-codinglab commented:
Approved
